### ISSUE TITLE
Mismatch with the corresponding paper

### ISSUE THREAD

**Issue by zhuyinheng** [mgharbi/hdrnet_legacy#28]
In the corresponding paper, section 3.4.1:

![screenshot from 2018-06-29 11 34 17](https://user-images.githubusercontent.com/16465741/42071913-868919d6-7b90-11e8-98cd-709ba3fcfd75.png)
as it's shown in the equation, (M_c)^T should be different for each c from 0 to 2 and so as ϕ_c

But when it comes to the implementation of this equation, which is in the HDRNetCurves._guide() in [model.py](https://github.com/mgharbi/hdrnet_legacy/blob/master/hdrnet/models.py)

```{python}
 # Color space change
    idtity = np.identity(nchans, dtype=np.float32) + np.random.randn(1).astype(np.float32)*1e-4
    ccm = tf.get_variable('ccm', dtype=tf.float32, initializer=idtity)
    with tf.name_scope('ccm'):
        ccm_bias = tf.get_variable('ccm_bias', shape=[nchans,], dtype=tf.float32, initializer=tf.constant_initializer(0.0))

        guidemap = tf.matmul(tf.reshape(input_tensor, [-1, nchans]), ccm)
        guidemap = tf.nn.bias_add(guidemap, ccm_bias, name='ccm_bias_add')

        guidemap = tf.reshape(guidemap, tf.shape(input_tensor))
```

It seems that there is only one M and ϕ, or in another word M_0=M_1=M_2 and ϕ_0=ϕ_1=ϕ_2.
Is the paper right or the implementation? @mgharbi

**Comment by mgharbi** [comment]
C is the channel index in [0,2] there are nchans=3 of them

**Comment by mgharbi** [comment]
Mc are the rows of ccm with dim 1x3

**Comment by mgharbi** [comment]
Mc are the rows of ccm with dim 1x3

**Comment by zhuyinheng** [comment]
Sorry about that, it seems I haven't fully understand the paper and misunderstood this. 
Thanks for reply!

**Comment by ColdCodeCool** [comment]
@mgharbi  hi, Michael, I think the pixel at (x,y) of input ϕ should not be indexed with c. Just as you mentioned McT is a row of ccm, so it is a vector, then ϕ should be a vector with 3 dimensions (r, g, b), then g(x,y) will be a scalar, otherwise it will be a vector. How do you think?

**Comment by mgharbi** [comment]
I understand now. You are correct, there is only one feature vector \phi and the subscript for \phi is a typo. _c indexes the output channels after the color transform.
